devito/arch/compiler.py

```diff
@@ -456,6 +456,20 @@ def __lookup_cmds__(self):
         self.MPICXX = 'mpicxx'
 
 
+class DPCPPCompiler(Compiler):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, cpp=False, **kwargs)
```

Comment:
@reguly shouldn't this be `cpp=True` ?

Reply:
No, I found the icx compiler packaged with dpcpp which is willing to compile C code. Though I guess cpp=False is the default, so this is unnecessary?

Reply:
yep probably unnecessary, but would also be useful if you could add a comment about this:

> I found the icx compiler packaged with dpcpp which is willing to compile C code.

This is news to me -- I thought it expected cpp (hence the compiler name...)

Reply:
Added some comments, hope that clarifies this. Intel now has too many compilers! :)